### DISCUSSION TITLE
Just an observation for eco_charge mode

### DISCUSSION THREAD

**Discussion by lampra1** [mletenay/home-assistant-goodwe-inverter#216] [Show and tell]
My goodwe battery is 13.1kwh and the inverter is ET plus 5kw.
It seems to me that the battery SOC is estimated based on battery voltage. My observation is that, when inverter mode is set to eco_charge, the battery never reaches 100% SOC. Because of a surge of the voltage after 90% SOC the BMS disables charging. This happens at eco_charge mode, both at 0 or at higher values.
Though this is not the case at eco_discharge mode at 0 which charges the battery to 100% SOC.

**Comment by mletenay** [comment]
Well, HA just reads values from inverter, there is no calculation or logic.
What happens, happens inside the inverter's firmware ...
